### ISSUE TITLE
assignment to non-final `@Owning` field allowed even when enclosing class is `@MustCall("")`

### ISSUE THREAD

**Issue by msridhar** [kelloggm/object-construction-checker#366]
Here is the test case:

```java
class Enclosing {
    @MustCall("a")
    static class Foo {
        void a() {}
    }
    static class Nested {
        @Owning Foo foo;
        @CreatesObligation("this")
        void initFoo() {
            if (this.foo == null) {
                // :: error: required.method.not.called
                this.foo = new Foo();
            }
        }
    }
}
```

I've commited a variant of this test on [a branch](https://github.com/kelloggm/object-construction-checker/tree/non-final-owning-no-mustcall).  No error is reported at the assignment to `this.foo`, even though the `Nested` class is `@MustCall("")`.

I kind of feel like in this case, we should report at error at the declaration of field `foo` rather than at every write to `foo`; but not 100% sure.

**Comment by msridhar** [comment]
@Nargeshdb @kelloggm

**Comment by msridhar** [comment]
@kelloggm we can close this too, yes?

**Comment by kelloggm** [comment]
Closed by #388.
